code401challenges/src/main/java/code401challenges/stacksandqueues/PseudoQueue.java

```diff
@@ -0,0 +1,25 @@
+package code401challenges.stacksandqueues;
+
+public class PseudoQueue <T>{
+
+    protected Stack<T> stack1 = new Stack<>();
+    protected Stack<T> stack2 = new Stack<>();
+
+    public void enqueue(T value) {
+
+        stack1.push(value);
+    }
+
+    public Node<T> dequeue() {
+        if(stack2.getTop() == null) {
+            while(stack1.getTop() != null) {
+                stack2.push((T) stack1.pop());
+            }
+        } else {
+            stack2.pop();
+        }
+
+        return (Node<T>) stack2.peek().getValue();
```

Comment:
I don't think you want to cast this into a `Node<T>`, you want to just return the value